src/components/TextField/TextField.jsx

```diff
@@ -2,15 +2,24 @@ import React from 'react';
 import styles from './TextField.styl';
 import PropTypes from 'prop-types';
 
-const noop = () => {};
+const noop = () => { };
 
-const TextField = ({ placeholder, id, val = null }) => {
-    return <input {...val} onChange={noop} id={id} className={styles.TextField} type='text' placeholder={placeholder} />;
+const TextField = ({ placeholder, id, val, onChange = noop, mapToStateName }) => {
+  return <input
+    {...(val || val === '' ? { value: val } : {})}
```

Comment:
it somewhat hard to figure out what's your intention here. Better to extract as a method